beater/beater.go

```diff
@@ -45,11 +45,14 @@ func (bt *ApmServer) Run(b *beat.Beat) error {
 		go bt.client.PublishAll(events)
 	}
 
-	bt.server, err = server.New(bt.config.Server)
-	if err != nil {
+	ucfg := bt.config.ApmServerConfig
+	apmServerConfig := server.DefaultConfig
+	if err := ucfg.Unpack(&apmServerConfig); err != nil {
```

Comment:
the common Config is used so that the Unpacking can happen in the package itself and the knowledge about the config must not be shared outside. Above you access `server.DefaultConfig` which could also be private in the package. This also prevents potential circular imports and coupling between the beater and the server package.